src/core/annotation.js

```diff
@@ -1359,26 +1369,41 @@ class WidgetAnnotation extends Annotation {
     return [initialState.font, initialState.fontName, initialState.fontSize];
   }
 
-  _computeFontSize(font, fontName, fontSize, height) {
+  _computeFontSize(font, fontName, fontSize, height, lineCount) {
     if (fontSize === null || fontSize === 0) {
-      const em = font.charsToGlyphs("M")[0].width / 1000;
-      // According to https://en.wikipedia.org/wiki/Em_(typography)
-      // an average cap height should be 70% of 1em
-      const capHeight = 0.7 * em;
-      // 1.5 * capHeight * fontSize seems to be a good value for lineHeight
-      fontSize = Math.max(1, Math.floor(height / (1.5 * capHeight)));
+      if (lineCount === -1) {
+        fontSize = Math.round(0.8 * height * 10) / 10;
+      } else {
+        // Hard to guess how many lines there are
+        // let's say that fonsize is 10.
+        fontSize = 10;
+        let lineHeight = fontSize / 0.8;
+        let numberOfLines = Math.round(height / lineHeight);
+        // multiline so there are at least 2 lines
+        numberOfLines = Math.max(2, numberOfLines, lineCount);
+        // finally we have lineHeight
```

Comment:
This comment and the one below can be removed as the code is clear enough here so that these comments don't add anything.